File: composeApp/src/wasmJsMain/kotlin/com/gleb/designer/web/app/pages/HomePage.kt
package com.gleb.designer.web.app.pages

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.UriHandler
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.gleb.designer.web.app.components.TextButton
import com.gleb.designer.web.app.theme.Strings
import com.gleb.designer.web.app.theme.ThemeColors
import com.seiko.imageloader.rememberImagePainter
import designerwebsite.composeapp.generated.resources.Res
import designerwebsite.composeapp.generated.resources.emoji
import designerwebsite.composeapp.generated.resources.home_page_background
import org.jetbrains.compose.resources.painterResource

@Composable
fun HomePage(
    strings: Strings,
    isBlackTheme: Boolean,
    themeColors: ThemeColors,
    uriHandler: UriHandler
) {
    Box(
        modifier = Modifier.fillMaxWidth()
    ) {
        if (isBlackTheme) {
            Image(
                painter = painterResource(Res.drawable.home_page_background),
                contentDescription = null,
                contentScale = ContentScale.FillWidth,
                modifier = Modifier.fillMaxWidth().align(Alignment.Center)
            )
        }
        Column(
            modifier = Modifier.fillMaxWidth().padding(125.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Image(
                painter = rememberImagePainter("https://raw.githubusercontent.com/GlebPoroshin/DesignerWebApp/main/composeApp/src/commonMain/composeResources/drawable/emoji.png"),
                contentDescription = null,
                modifier = Modifier.size(240.dp).clip(CircleShape)
            )
            Text(
                text = strings.name,
                style = TextStyle(
                    fontSize = 64.sp,
                    fontWeight = FontWeight.Bold,
                    color = themeColors.thirdColor
                )
            )
            Spacer(Modifier.height(16.dp))
            Text(
                text = strings.profession,
                style = TextStyle(
                    fontSize = 24.sp,
                    fontWeight = FontWeight.SemiBold,
                    brush = themeColors.blueToWhiteGradient
                )
            )
            Spacer(Modifier.height(8.dp))
            Text(
                text = strings.shortDescription,
                style = TextStyle(
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Medium,
                    color = themeColors.thirdColor
                )
            )
            Spacer(Modifier.height(40.dp))
            TextButton(
                modifier = Modifier.width(180.dp),
                text = strings.contactMe,
                textColor = themeColors.onMainColor,
                borderColor = themeColors.thirdColor,
                shape = RoundedCornerShape(72.dp),
                onClick = { uriHandler.openUri("https://t.me/zipzen") }
            )
        }
    }
}

File: react-portfolio/src/pages/HomePage.js
import React from 'react';
import styled from 'styled-components';
import Button from '../components/Button';
import ImageWithShimmer from '../components/ImageWithShimmer';

const HomeContainer = styled.section`
  min-height: 80vh;
  display: flex;
  align-items: center;
  
  @media (max-width: 768px) {
    flex-direction: column;
    text-align: center;
    padding: 2rem 0;
  }
`;

const ContentContainer = styled.div`
  flex: 1;
  padding-right: 2rem;
  
  @media (max-width: 768px) {
    padding-right: 0;
    margin-bottom: 2rem;
  }
`;

const ImageContainer = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  
  @media (max-width: 768px) {
    width: 100%;
  }
`;

const NameHeading = styled.h1`
  font-size: 3rem;
  font-weight: 700;
  margin-bottom: 0.5rem;
  color: ${props => props.theme.thirdColor};
  
  @media (max-width: 768px) {
    font-size: 2.5rem;
  }
`;

const Profession = styled.h2`
  font-size: 1.5rem;
  font-weight: 500;
  margin-bottom: 1.5rem;
  color: ${props => props.theme.fourthColor};
  
  @media (max-width: 768px) {
    font-size: 1.25rem;
  }
`;

const Description = styled.p`
  font-size: 1.1rem;
  margin-bottom: 2rem;
  line-height: 1.6;
  color: ${props => props.theme.fourthColor};
  max-width: 600px;
  text-align: justify;
  
  @media (max-width: 768px) {
    font-size: 1rem;
    text-align: center;
  }
`;

const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;
  
  @media (max-width: 768px) {
    justify-content: center;
  }
`;

const HomePage = ({ strings, uriHandler }) => {
  const handleContactClick = () => {
    document.getElementById('contacts').scrollIntoView({ behavior: 'smooth' });
  };

  return (
    <HomeContainer id="home">
      <ContentContainer>
        <NameHeading>{strings.name}</NameHeading>
        <Profession>{strings.profession}</Profession>
        <Description>{strings.shortDescription}</Description>
        <ButtonContainer>
          <Button primary onClick={handleContactClick}>
            {strings.contactMe}
          </Button>
        </ButtonContainer>
      </ContentContainer>
      <ImageContainer>
        <ImageWithShimmer
          src="https://raw.githubusercontent.com/GlebPoroshin/DesignerWebApp/react/composeApp/src/commonMain/composeResources/drawable/emoji.png"
          alt="Emoji Sticker"
          width="240px"
          height="240px"
          objectFit="contain"
        />
      </ImageContainer>
    </HomeContainer>
  );
};

export default HomePage; 